exotica_core/include/exotica_core/problems/dynamic_time_indexed_shooting_problem.h

```diff
@@ -53,9 +54,11 @@ class DynamicTimeIndexedShootingProblem : public PlanningProblem, public Instant
     double get_tau() const;  ///< Returns the discretization timestep tau
 
     Eigen::MatrixXd get_X() const;             ///< Returns the state trajectory X
+    Eigen::MatrixXd get_X(int i) const;        ///< Returns the state at time t
```

Comment:
I am not sure I am a fan of this overload - isn't `get_X().col(t)` an option?

Reply:
It is, but since it's not a MatrixXdRef, won't it be copying the entire trajectory every time?

Reply:
Okay, good point! Still required to change to `int t`